cudaaligner/src/myers_gpu.cu

```diff
@@ -335,28 +336,33 @@ int32_t myers_compute_edit_distance(std::string const& target, std::string const
     if (get_size(query) == 0)
         return get_size(target);
 
+    const int32_t n_words = (get_size(query) + word_size - 1) / word_size;
+    matrix<int32_t> score_host;
+
     cudaStream_t stream;
     CGA_CU_CHECK_ERR(cudaStreamCreate(&stream));
+    {
+        DefaultDeviceAllocator allocator = create_default_device_allocator();
 
-    int32_t max_sequence_length = std::max(get_size(target), get_size(query));
-    device_buffer<char> sequences_d(2 * max_sequence_length);
-    device_buffer<int32_t> sequence_lengths_d(2);
+        int32_t max_sequence_length = std::max(get_size(target), get_size(query));
+        device_buffer<char> sequences_d(2 * max_sequence_length, allocator, stream);
+        device_buffer<int32_t> sequence_lengths_d(2, allocator, stream);
 
-    const int32_t n_words = (get_size(query) + word_size - 1) / word_size;
-    batched_device_matrices<myers::WordType> pv(1, n_words * (get_size(target) + 1), stream);
-    batched_device_matrices<myers::WordType> mv(1, n_words * (get_size(target) + 1), stream);
-    batched_device_matrices<int32_t> score(1, n_words * (get_size(target) + 1), stream);
-    batched_device_matrices<myers::WordType> query_patterns(1, n_words * 4, stream);
+        batched_device_matrices<myers::WordType> pv(1, n_words * (get_size(target) + 1), allocator, stream);
+        batched_device_matrices<myers::WordType> mv(1, n_words * (get_size(target) + 1), allocator, stream);
+        batched_device_matrices<int32_t> score(1, n_words * (get_size(target) + 1), allocator, stream);
+        batched_device_matrices<myers::WordType> query_patterns(1, n_words * 4, allocator, stream);
 
-    std::array<int32_t, 2> lengths = {static_cast<int32_t>(get_size(query)), static_cast<int32_t>(get_size(target))};
-    CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequences_d.data(), query.data(), sizeof(char) * get_size(query), cudaMemcpyHostToDevice, stream));
-    CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequences_d.data() + max_sequence_length, target.data(), sizeof(char) * get_size(target), cudaMemcpyHostToDevice, stream));
-    CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequence_lengths_d.data(), lengths.data(), sizeof(int32_t) * 2, cudaMemcpyHostToDevice, stream));
+        std::array<int32_t, 2> lengths = {static_cast<int32_t>(get_size(query)), static_cast<int32_t>(get_size(target))};
+        CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequences_d.data(), query.data(), sizeof(char) * get_size(query), cudaMemcpyHostToDevice, stream));
```

Comment:
Again, can this be `device_copy_n`?

Reply:
Good point, there are probably more locations where cudaMemcpy can be replaced in cudaaligner. Will address that in a separate PR. Opened ticket #355.

Reply:
can I resolve this for now?

Reply:
marked as resolved